packages/reschema/src/ReSchema.re

```diff
@@ -5,30 +5,37 @@ module type Lenses = {
   let get: (state, field('a)) => 'a;
 };
 
+type childFieldError = {
+  error: string,
+  index: int,
+  name: string,
```

Comment:
hmm wonder if we could fit into the abstraction a `'a` kind of type to let the users pass the actual GADT constructor, still thinking here